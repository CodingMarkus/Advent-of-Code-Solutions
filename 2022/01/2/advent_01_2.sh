#!/bin/sh

# Run as:
# cat advent_01_2_input.txt | sh advent_01_2.sh

set -e

sumUp()
(
	current=0
	while read -r line
	do
		if [ -z "$line" ]
		then
			echo $current
			current=0
		else
			current=$(( current + line ))
		fi
	done
	printf '%s\n' "$current"
)

total=0

for top3 in $( sumUp | sort -n | tail -n 3 )
do
	total=$(( total + top3 ))
done

echo $total