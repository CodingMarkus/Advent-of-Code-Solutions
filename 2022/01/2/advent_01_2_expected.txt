Answer: 45000
Answer: 202585
